Refit/RequestBuilderImplementation.cs

```diff
@@ -446,6 +463,13 @@ public RestMethodInfo(Type targetInterface, MethodInfo methodInfo, RefitSettings
             Name = methodInfo.Name;
             MethodInfo = methodInfo;
 
+            var uma = methodInfo.GetCustomAttributes(true)
+                .OfType<UniqueNameAttribute>()
+                .FirstOrDefault();
+
+            if (uma != null)
+                Name = uma.MethodName;
```

Comment:
Yes you are right and I wish there was another way, but it is almost impossible to get the same method signature in runtime as during stub generation.